3rdParty/LightPcapNg/LightPcapNg/src/light_alloc.c

```diff
@@ -54,8 +54,12 @@ light_pcapng light_alloc_block(uint32_t block_type, const uint32_t *block_body,
 
 	PADD32(block_body_length, &actual_size);
 
-	pcapng_block->block_total_length = actual_size; // This value MUST be a multiple of 4.
-	block_body_size = actual_size - 2 * sizeof(pcapng_block->block_total_length) - sizeof(pcapng_block->block_type);
+	pcapng_block->block_total_length = actual_size + 2 * sizeof(pcapng_block->block_total_length) + sizeof(pcapng_block->block_type); // This value MUST be a multiple of 4.
+	block_body_size = actual_size;
+	
+	// Test passes with the old code???
```

Comment:
I highlight here, by using the old version code, the test can pass, however, the latest code will fail with the following error.

```
ERROR at /home/mujin/PcapPlusPlus/3rdParty/LightPcapNg/LightPcapNg/src/light_pcapng.c::__parse_options::67: 152 != 0
ERROR at /home/mujin/PcapPlusPlus/3rdParty/LightPcapNg/LightPcapNg/src/light_pcapng.c::__parse_options::67: 152 != 0
TestPcapNgFileReadWriteAdv         : FAILED (/home/mujin/PcapPlusPlus/Tests/Pcap++Test/Tests/FileTests.cpp:521). Assert EQUAL failed:
   Actual:   
   Expected: Linux 3.18.1-1-ARCH
```

Reply:
It's related to this PR: https://github.com/rvelea/LightPcapNg/pull/6/files

Reply:
I have no idea why they made this change, but the pcapng file in Wireshark shows the expected OS information. Maybe they have a bug? 🤔 

![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/1351dcf1-ff72-47d4-9c50-0decc9941922)

Reply:
@seladb  I decided to keep the old implementation.

Reply:
please check the comment: https://github.com/seladb/PcapPlusPlus/pull/1336#discussion_r1513971881